File: coco/src/main/java/coco/ep/m4/Exp.java
package coco.ep.m4;

import java.util.List;

public interface Exp<FT> extends coco.ep.m2.Exp<FT>, Factory<FT> {

    coco.ep.Exp<FT> simplify();
    void truncate(int level);

    List<Double> collect();
}

File: coco_csharp/CoCo/EP/M4/Exp.cs
using System.Collections.Generic;

namespace EP.M4 {
    public interface Exp<FT> : Factory<FT>, EP.M2.Exp<FT>  {
        public EP.Exp<FT> simplify();
        public void truncate(int level);
        public List<double> collect();
    }
}